src/ser/serde.rs

```diff
@@ -53,22 +56,54 @@ impl Serialize for Bson {
     where
         S: ser::Serializer,
     {
-        match *self {
```

Comment:
this was updated to just call through to the underlying value's serialize method. Previously, it was the other way around. This lets us use named structs in the data model instead of nameless maps.